tests/functional/includes/Standard_Source_Cest.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+namespace ISC\Tests\Functional;
+
+/**
+ * Tests for ISC\Standard_Source to handle the standard source text
+ */
+class Standard_Source_Cest {
+
+	public function _before(\FunctionalTester $I) {
+		// enable the overlay
+		$existingOption = $I->grabOptionFromDatabase('isc_options');
+		$existingOption['display_type'] = ['overlay'];
+		// Make sure we enable the "custom_text" standard source with the text "@ ISC"
+		$existingOption['standard_source'] = 'custom_text';
+		$existingOption['standard_source_text'] = '@ ISC';
+
+		// Update the option in the database with the new array.
+		$I->haveOptionInDatabase( 'isc_options', $existingOption );
+
+		// prepare an image
+		$I->havePostInDatabase( [
+           'ID' => 123,
+           'post_title' => 'Image One',
+           'guid' => 'https://example.com/image-one.jpg',
+       ] );
+		// image has a source by default
+		$I->havePostmetaInDatabase( 123, 'isc_image_source', 'Author A' );
+	}
+
+	/**
+	 * Setup:
+	 * - image has a source
+	 * - image is set to use the standard source
+	 * => show the standard source text "@ ISC"
+	 *
+	 * @param \FunctionalTester $I
+	 *
+	 * @return void
+	 */
+	public function test_standard_custom_text( \FunctionalTester $I ) {
+		// set the image to use the standard source
+		$I->havePostmetaInDatabase( 123, 'isc_image_source_own', 1 );
+		$I->havePageInDatabase( [
+			'post_name'    => 'test-page',
+			'post_content' => '<img src="https://example.com/image-one.jpg" />',
+		] );
+
+		// Go to the page.
+		$I->amOnPage( '/test-page' );
+		$I->seeInSource( '<span id="isc_attachment_123" class="isc-source "><img decoding="async" src="https://example.com/image-one.jpg" /><span class="isc-source-text">Quelle: @ ISC</span></span>' );
+	}
+
+	/**
+	 * Setup:
+	 * - image has no source
+	 * - image is set to use the standard source
+	 * => show the standard source text "@ ISC"
+	 */
+	public function test_standard_custom_text_no_source( \FunctionalTester $I ) {
+		// set the image to use the standard source
+		$I->havePostmetaInDatabase( 123, 'isc_image_source_own', 1 );
+		// remove the explicit image source
+		$I->dontHavePostMetaInDatabase( [ 'post_id' => 123, 'meta_key' => 'isc_image_source' ] );
+
+		$I->havePageInDatabase( [
+			'post_name'    => 'test-page',
+			'post_content' => '<img src="https://example.com/image-one.jpg" />',
+		] );
+
+		// Go to the page.
+		$I->amOnPage( '/test-page' );
+		// the actual HTML output, including the data attribute and the overlay code at the beginning of the DIV container
+		$I->seeInSource( '<span id="isc_attachment_123" class="isc-source "><img decoding="async" src="https://example.com/image-one.jpg" /><span class="isc-source-text">Quelle: @ ISC</span></span>' );
+	}
```

Comment:
**Suggestion:** It's a good practice to use data providers in tests when you have the same set of tests but with different data sets. It makes your tests cleaner and easier to maintain. [best practice]
```suggestion
 /**
  * @dataProvider standardSourceProvider
  */
 public function test_standard_custom_text( \FunctionalTester $I, \Codeception\Example $example ) {
     ...
     // use $example['source'] and $example['expected'] in your test
 }

 /**
  * @return array
  */
 protected function standardSourceProvider(): array
 {
     return [
         ['source' => ..., 'expected' => ...],
         ['source' => ..., 'expected' => ...],
         // add more data sets
     ];
 }
```